scipy/optimize/tests/test_linprog.py

```diff
@@ -1567,16 +1572,40 @@ def test_callback(self):
         _assert_success(res, desired_fun=-18.0, desired_x=[2, 6])
 
     @pytest.mark.parametrize("options",
-                             [{"dual_feasibility_tolerance": -1},
+                             [{"maxiter": -1},
+                              {"disp": -1},
+                              {"presolve": -1},
+                              {"time_limit": -1},
+                              {"dual_feasibility_tolerance": -1},
                               {"primal_feasibility_tolerance": -1},
-                              {"simplex_dual_edge_weight_strategy": 10},
+                              {"ipm_optimality_tolerance": -1},
+                              {"simplex_dual_edge_weight_strategy": "ekki"},
                               ])
     def test_invalid_option_values(self, options):
         def f(options):
             linprog(1, method=self.method, options=options)
         options.update(self.options)
         assert_warns(OptimizeWarning, f, options=options)
 
+    def test_optimize_result(self):
+        # check all fields in OptimizeResult
+        c, A_ub, b_ub, A_eq, b_eq, bounds = very_random_gen(0)
+        res = linprog(c, A_ub=A_ub, b_ub=b_ub, A_eq=A_eq, b_eq=b_eq,
+                      bounds=bounds, method=self.method, options=self.options)
+        assert_(res.success)
+        assert_(res.nit)
+        assert_(not res.status)
+        assert_(res.message == "Optimization terminated successfully.")
+        assert_allclose(c @ res.x, res.fun)
+        assert_allclose(b_eq - A_eq @ res.x, res.con, atol=1e-12)
+        assert_allclose(b_ub - A_ub @ res.x, res.slack, atol=1e-12)
+        assert_equal(res.method, self.method)
+        # crossover_nit is nonzero only for highs-ipm
+        assert_equal(res.crossover_nit == 0, self.method != "highs-ipm")
```

Comment:
This test is currently failing for IPM - `res.crossover_nit` is always 0.

Reply:
I'll take a look and see if this value is being set properly

Reply:
@mdhaber as far as I can tell this is working properly, there are no crossover iterations reported by HiGHS for this problem.  I've also tried a bunch of different seeds in the `very_random_gen(seed)` (after fixing it to use the seed) and still can't find one that does any crossover

Reply:
Hmm ok. Is there another problem for which you know there are crossover iterations?

Reply:
I'm looking around for one right now.  Another strategy might be to set the solution tolerances very low?

Reply:
Thanks. Sorry I didn't just pick the right problem to begin with.